In [0]:
import zipfile
from google.colab import drive
drive.mount("/content/drive/")

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/drive/


In [0]:
#!wget "/content/drive/My Drive/IntroML-Project-SP20/"
!unzip "/content/drive/My Drive/IntroML-Project-SP20/common-voice/cv-valid-train.zip" -d "drive/My Drive/IntroML-Project-SP20/common-voice/train"

Archive:  /content/drive/My Drive/IntroML-Project-SP20/common-voice/cv-valid-train.zip
warning [/content/drive/My Drive/IntroML-Project-SP20/common-voice/cv-valid-train.zip]:  2482475816 extra bytes at beginning or within zipfile
  (attempting to process anyway)
error [/content/drive/My Drive/IntroML-Project-SP20/common-voice/cv-valid-train.zip]:  start of central directory not found;
  zipfile corrupt.
  (please check that you have transferred or created the zipfile in the
  appropriate BINARY mode and that you have compiled UnZip properly)


In [0]:
!ls "/content/drive/My Drive/IntroML-Project-SP20/common-voice/dev/"

In [0]:
from google.colab import files
uploaded = files.upload()

Saving cv-valid-dev.csv to cv-valid-dev (1).csv
Saving cv-valid-test.csv to cv-valid-test.csv
Saving cv-valid-train.csv to cv-valid-train.csv


In [0]:
import io
df1 = pd.read_csv(io.BytesIO(uploaded['cv-valid-dev.csv']), delimiter=",")
df2 = pd.read_csv(io.BytesIO(uploaded['cv-valid-train.csv']), delimiter=",")
df3 = pd.read_csv(io.BytesIO(uploaded['cv-valid-test.csv']), delimiter=",")

In [0]:
#df1.head(5)

Separate accented data from the data that has NULL entries for accent.

Naming and organization convention as well as data format can be found here: https://www.kaggle.com/mozillaorg/common-voice

In [0]:
import os
import os.path
import numpy as np
import pandas as pd
from sklearn.preprocessing import StandardScaler
import matplotlib.pyplot as plt

%matplotlib inline

In [0]:
def filecount(dir_name):
    """This function counts number of files in directory provided"""
    return len([f for f in os.listdir(dir_name) if os.path.isfile(f)])

In [0]:
# Insert folder path
data_dir = "/"

In [0]:
# dev data
#df1 = pd.read_csv("../data/cv-valid-dev.csv", delimiter=",")
df1.dataframeName = "cv-valid-dev.csv"
row_df1, col_df1 = df1.shape
print("There are %d rows and %d columns in cv-valid-dev.csv" %(row_df1, col_df1))

# train data
#df2 = pd.read_csv("../data/cv-valid-train.csv", delimiter=",")
df2.dataframeName = "cv-valid-train.csv"
row_df2, col_df2 = df2.shape
print("There are %d rows and %d columns in cv-valid-train.csv" %(row_df2, col_df2))

# test data
#df3 = pd.read_csv("../data/cv-valid-test.csv", delimiter=",")
df3.dataframeName = "cv-valid-test.csv"
row_df3, col_df3 = df3.shape
print("There are %d rows and %d columns in cv-valid-test.csv" %(row_df3, col_df3))

There are 4076 rows and 8 columns in cv-valid-dev.csv
There are 195776 rows and 8 columns in cv-valid-train.csv
There are 3995 rows and 8 columns in cv-valid-test.csv


There are 4076 rows and 8 columns in cv-valid-dev.csv

There are 195776 rows and 8 columns in cv-valid-train.csv

There are 3995 rows and 8 columns in cv-valid-test.csv

In [0]:
# separate accented data
# dev data
accent_true = df1["accent"].notna()
df_accent_dev = df1[accent_true]
row_acc_dev, col_acc_dev = df_accent_dev.shape
df_accent_dev.to_csv("cv-valid-dev-acc.csv")
print("There are %d rows and %d columns in cv-valid-dev-acc.csv" %(row_acc_dev, col_acc_dev))

# training data
accent_true = df2["accent"].notna()
df_accent_train = df2[accent_true]
row_acc_tr, col_acc_tr = df_accent_train.shape
df_accent_train.to_csv("cv-valid-train-acc.csv")
print("There are %d rows and %d columns in cv-valid-train-acc.csv" %(row_acc_tr, col_acc_tr))

# test data
accent_true = df3["accent"].notna()
df_accent_test = df3[accent_true]
row_acc_test, col_acc_test = df_accent_test.shape
df_accent_test.to_csv("cv-valid-test-acc.csv")
print("There are %d rows and %d columns in cv-valid-test-acc.csv" %(row_acc_test, col_acc_test))


There are 1350 rows and 8 columns in cv-valid-dev-acc.csv
There are 64711 rows and 8 columns in cv-valid-train-acc.csv
There are 1338 rows and 8 columns in cv-valid-test-acc.csv


In [0]:
df_accent_dev.head(5)

filename  ... duration
4   cv-valid-dev/sample-000004.mp3  ...      NaN
5   cv-valid-dev/sample-000005.mp3  ...      NaN
8   cv-valid-dev/sample-000008.mp3  ...      NaN
9   cv-valid-dev/sample-000009.mp3  ...      NaN
13  cv-valid-dev/sample-000013.mp3  ...      NaN

[5 rows x 8 columns]

In [0]:
df_accent_train.head(5)

filename  ... duration
5   cv-valid-train/sample-000005.mp3  ...      NaN
8   cv-valid-train/sample-000008.mp3  ...      NaN
13  cv-valid-train/sample-000013.mp3  ...      NaN
14  cv-valid-train/sample-000014.mp3  ...      NaN
19  cv-valid-train/sample-000019.mp3  ...      NaN

[5 rows x 8 columns]

In [0]:
df_accent_test.head(5)

filename  ... duration
3   cv-valid-test/sample-000003.mp3  ...      NaN
5   cv-valid-test/sample-000005.mp3  ...      NaN
8   cv-valid-test/sample-000008.mp3  ...      NaN
9   cv-valid-test/sample-000009.mp3  ...      NaN
14  cv-valid-test/sample-000014.mp3  ...      NaN

[5 rows x 8 columns]

In [0]:
from google.colab import files
files.upload()

Saving kaggle.json to kaggle.json


{'kaggle.json': b'{"username":"mariamelgamal","key":"f2e3d41c647d4c53fa67fde21f76301c"}'}

In [0]:
!pip install -q kaggle
!mkdir -p ~/.kaggle
!cp kaggle.json ~/.kaggle/
!ls ~/.kaggle
!chmod 600 /root/.kaggle/kaggle.json  # set permission

kaggle.json


In [0]:
!kaggle datasets download -d mozillaorg/common-voice -p /content/drive/My\ Drive/IntroML-Project-SP20/common-voice

100% 12.0G/12.0G [03:54<00:00, 60.1MB/s]
100% 12.0G/12.0G [03:54<00:00, 55.1MB/s]
